src/rustdoc_cmd.rs

```diff
@@ -129,36 +130,43 @@ impl RustdocCommand {
         if !self.deps {
             cmd.arg("--no-deps");
         }
-        if config.is_stderr_tty() {
-            cmd.arg("--color=always");
-        }
+
+        // TODO
+        let color_choice = match anstream::stderr().current_choice() {
+            ColorChoice::Always | ColorChoice::AlwaysAnsi => "--color=always",
+            ColorChoice::Auto if anstream::stderr().is_terminal() => "--color=always",
+            ColorChoice::Auto | ColorChoice::Never => "--color=never",
+        };
+
+        cmd.arg(color_choice);
+        // if config.is_stderr_tty() {
+        //    cmd.arg("--color=always");
+        // }
```

Comment:
I updated this so it reads the current stderr configuration.  Let me know if the documentation isn't clear or if this is too "hacky".  Right now, we don't have a way to read the current color choice in our config API, but I can add that if we want it.  However, because we write the child process stderr (which has colors) to our `AutoStream` that strips colors if we indicate our preference for that, we will get the correct color preferences here.  I can add a test here for if you think that's a good idea.

Reply:
If you don't mind writing a test, I'd love to have one! `AutoStream`'s behavior is nice but a bit surprising, so it'd be nice to have a test we can point future readers of this code to (including me).

It would be especially useful if in the future we yet again switch libraries, and the next library's stream implementation doesn't do this kind of magically-automatic ANSI code stripping.

Reply:
This is a little more nebulous than I thought - it works when `RustdocCommand::silence` is true, but when it is false, since we just forward the rustdoc err stream to stderr (actual tty stderr, not `GlobalConfig::stderr`), we're not currently using the configured stderr for the rustdoc progress stream.  We can redirect it using `Command::spawn` and a thread that acts as a pipe between the child process stderr and our stderr if we want.  Do you think a library user would expect this behavior? That is, would they expect the rustdoc progress output to go through the `cargo-semver-checks`' configured stderr.  I'm leaning towards yes, but it will add more complexity to the function because we have to use threads.

By the way, thanks for all your patience and great feedback on this PR!

Reply:
I think definitely not in this PR, and possibly not at all. We'll have to solicit feedback from users, and I'm wary of taking on even more technical complexity when we aren't sure users want the feature it produces.

Maybe open an issue describing the concern and the nature of the decision point here, and we revisit it in the future. Otherwise this PR for switching to a different stream coloring library is going to end up being a giant revamp of everything, and it'll take forever to merge.

Reply:
Also, thank you for all the care and skill you are investing into cargo-semver-checks! It's my pleasure to help as best I can, and it's lovely to work with someone as thoughtful and kind as you!

Reply:
Got it, that makes a lot of sense.  I just re-added the color flag on our rustdoc invocation to respect the color preferences.  Do you still want a test for this? It's not using any of the `anstream` magic anymore.

Reply:
If we aren't materially changing the code here, it's fine as it was without a test.

But if you have a clever idea for how to elegantly write a test for something that previously wasn't tested, I wouldn't say no!

Reply:
I think the `generate_rustdoc` function is too long and does too much to be able to effectively test something like this.  It might be worth breaking it up in the future so we can write tests that make these behaviors more visible and guaranteed.  Since the behavior stays the same here, I don't know if it's worth writing the test in this already-big PR, but I'd be open to refactoring and writing it in the future.